### PR TITLE
[Refactor] Do not expose internal function in field, exception, expr, any_array , _ndrange, _ndarray

### PR DESCRIPTION
**Pull request by mzmzm**
Related issue = #3782

### CODE REVIEW

File: python/taichi/lang/_ndrange.py [hunk 1]
```diff
@@ -49,3 +49,6 @@ def __init__(self, r):
     def __iter__(self):
         for ind in self.r:
             yield _IntermediateMatrix(len(ind), 1, list(ind))
+
+
+__all__ = ['ndrange']
```
Comment (k-ye) [inline python/taichi/lang/_ndrange.py:54]: OOC, can we make `ndrange` a function? (It's a code smell issue: class should capitalize its first letter)

Comment (mzmzm) [reply python/taichi/lang/_ndrange.py:54]: Do you mean to add a new API `ndrange` and change `class ndrange` to `class Ndrange`? If so, we have `ti.ndrange` and `ti.Ndrange` to use.

Comment (k-ye) [reply python/taichi/lang/_ndrange.py:54]: I mean can we provide `ndrange` as a function, which can be implemented using some internal helper class like `class _Ndrange`. Users would still only need to use `ti.ndrange`

Comment (k-ye) [reply python/taichi/lang/_ndrange.py:58]: Anyway, that's a question for future works :-)

Comment (mzmzm) [reply python/taichi/lang/_ndrange.py:58]: f70cc8ca95706f8bd61c84b5330f8d9c8b5e880d

**Review by k-ye — APPROVED** [review]
lgtm!

**Pull request merged**